events_webhook/src/main/java/com/adobe/aio/event/webhook/service/EventVerifier.java

```diff
@@ -0,0 +1,126 @@
+   * @param url
+   */
+  public EventVerifier(String url) {
+    this.pubKeyService = new FeignPubKeyService(url);
+  }
+
+  /**
+   * Your Event Verifier instance must be created with the Adobe IO serving CDN host as below
+   */
+  public EventVerifier() {
+    this(ADOBE_IOEVENTS_SECURITY_DOMAIN);
+  }
+
+  /**
+   * Authenticate the event by checking the target recipient
+   * and verifying the signatures
+   * @param message - the event payload
+   * @param clientId - recipient client id in the payload
+   * @param headers - webhook request headers
+   * @return boolean - TRUE if valid event else FALSE
+   */
+  public boolean authenticateEvent(String message, String clientId, Map<String, String> headers) {
+    if (!isValidTargetRecipient(message, clientId)) {
+      logger.error("target recipient {} is not valid for message {}", clientId, message);
+      return false;
+    }
+    if (!verifyEventSignatures(message, headers)) {
+      logger.error("signatures are not valid for message {}", message);
+      return false;
+    }
+    return true;
+  }
+
+  private boolean verifyEventSignatures(String message, Map<String, String> headers) {
+    String[] digitalSignatures = {headers.get(ADOBE_IOEVENTS_DIGI_SIGN_1),
+        headers.get(ADOBE_IOEVENTS_DIGI_SIGN_2)};
+    String[] pubKeyPaths = {headers.get(ADOBE_IOEVENTS_PUB_KEY_1_PATH),
+        headers.get(ADOBE_IOEVENTS_PUB_KEY_2_PATH)};
+    // the signed payload is valid if any one of the signatures is valid
+    return verifySignature(message, pubKeyPaths[0], digitalSignatures[0]) ||
+        verifySignature(message, pubKeyPaths[1], digitalSignatures[1]);
+  }
+
+  private boolean verifySignature(String message, String publicKeyPath, String signature) {
+    byte[] data = message.getBytes(UTF_8);
+    try {
+      // signature generated at I/O Events side is Base64 encoded, so it must be decoded
+      byte[] sign = Base64.getDecoder().decode(signature);
+      Signature sig = Signature.getInstance("SHA256withRSA");
+      sig.initVerify(fetchPemEncodedPublicKey(publicKeyPath));
+      sig.update(data);
+      return sig.verify(sign);
+    } catch (GeneralSecurityException e) {
+      throw new AIOException("Error verifying signature for public key " + publicKeyPath
+          + ". Reason -> " + e.getMessage(), e);
+    }
+  }
+
+  private boolean isValidTargetRecipient(String message, String clientId) {
+    try {
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode jsonPayload = mapper.readTree(message);
+      JsonNode recipientClientIdNode = jsonPayload.get("recipient_client_id");
+      return (recipientClientIdNode != null && recipientClientIdNode.textValue() !=null
+          && recipientClientIdNode.textValue().equals(clientId));
+    } catch (JsonProcessingException e) {
+      throw new AIOException("error parsing the event payload during target recipient check..");
+    }
+  }
+
+  private PublicKey fetchPemEncodedPublicKey(String publicKeyPath) {
```

Comment:
as we have the new `pubKeyService` API this private method is of no use, let's delete it, wdyt ?

Reply:
ok

Reply:
done